integr8ly/integreatly-operator
# MGDAPI-582

# Description

https://issues.redhat.com/browse/MGDAPI-582


bump version of cro to include defaulting of cidr block if no param i…s provided to the config map

remove logic that was defaulting in integreatly

## Verification

Scenarios:
1) Verify fresh install by add on with cidr block passed -> the passed cidr block should be used by cro
2) Verify fresh install by add on with cidr block not passed -> cro should search for a default and find a suitable one and create a vpc
3) Verify upgrade -> the upgrade should succeed and no change to an existing vpc setup should occur.

**Steps taken to verify 1**

- install the add on passing a cidr value of 10.3.0.0/26
- update the image to `quay.io/laurafitzgerald/managed-api-service:v1.0.0` before rhoam pod is running
- Verify that secret `addon-managed-api-service-parameters` contains the cidr value
![Screenshot 2021-01-08 at 15 14 19](https://user-images.githubusercontent.com/6498727/104031129-451e8500-51c4-11eb-9e79-e2c4dbfa86f4.png)
- Verify that that is populated to the `cloud-resources-aws-strategies` config map
![Screenshot 2021-01-08 at 15 14 03](https://user-images.githubusercontent.com/6498727/104031171-4fd91a00-51c4-11eb-88f6-9dc0ee1dcd22.png)
- Verify that this entry is used to create the vpc in aws.
![Screenshot 2021-01-08 at 15 15 29](https://user-images.githubusercontent.com/6498727/104031223-5ebfcc80-51c4-11eb-926a-c0060c8844c0.png)

**Steps taken to verify 2** 

`INSTALLATION_TYPE=managed-api USE_CLUSTER_STORAGE=false make cluster/prepare/local`
`INSTALLATION_TYPE=managed-api USE_CLUSTER_STORAGE=false make code/run/service_account`

Verify `cloud-resources-aws-strategies` config map does not contain a default entry `{ "production": { "createStrategy": {} } }`
Verify that the vpc in aws has been created with a sensible default. 
Verify that the rds is created inside that vpc.

**Steps to verify 3**

- Install 1.0.1
- Do an upgrade -> As cro doesn't recreate any networking resources there should be no change to aws networking

## Type of change




## Checklist
